examples/streamgraph/streamgraph-example.js

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+import {stack as d3Stack, stackOffsetWiggle} from 'd3-shape';
+import {range, transpose} from 'd3-array';
+
+import {
+  XYPlot,
+  makeWidthFlexible,
+  AreaSeries
+} from 'react-vis';
+
+const FlexibleXYPlot = makeWidthFlexible(XYPlot);
+
+const NUMBER_OF_LAYERS = 20;
+const SAMPLES_PER_LAYER = 200;
+const BUMPS_PER_LAYER = 10;
+
+// Inspired by Bostock's version of Lee Byron’s test data generator.
+function bumps(samplesPerLayer, bumpsPerLayer) {
+  const dataOutline = (new Array(samplesPerLayer)).fill(0);
+  return range(bumpsPerLayer).reduce(res => bump(res, samplesPerLayer), dataOutline);
+}
+
+// bumpsPerLayer
+function bump(aggregatingData, samplesPerLayer) {
+  const x = 1 / (0.1 + Math.random());
+  const y = 2 * Math.random() - 0.5;
+  const z = 10 / (0.1 + Math.random());
+  for (let i = 0; i < samplesPerLayer; i++) {
```

Comment:
Is it necessary to mutate the elements of `aggregatingData` here, or can we simply return a new array?

```javascript
const bump = (aggregatingData, samplesPerLayer) => {
  const x = 1 / (0.1 + Math.random());
  const y = 2 * Math.random() - 0.5;
  const z = 10 / (0.1 + Math.random());

  return aggregatingData.map((v, i) => {
    const w = (i / samplesPerLayer - y) * z;
    return v + (x * Math.exp(-w * w));
  });
};
```